src/plugins/metrics_sender.go

```diff
@@ -98,36 +98,36 @@ func (r *MetricsSender) Process(msg *core.Message) {
 			}
 		}
 	} else if msg.Exact(core.AgentConfig) {
-		cmd, ok := msg.Data().(*proto.Command_AgentConfig)
-		if !ok {
-			log.Warnf("Failed to coerce Message to *proto.Command_AgentConfig: %v", msg.Data())
-			return
+		switch cmd := msg.Data().(type) {
+		case *proto.Command:
+			r.metricSenderBackoff(cmd.GetAgentConfig())
+		default:
+			log.Warnf("unsupported pause message %+v ", msg)
```

Comment:
Can you change the log message here to something like:
`log.Warnf("metrics sender expected %T type, but got: %T", &proto.Command{}, msg.Raw())`

Reply:
@dhurley sure. I have updated it.

I have `log.Warnf("metrics sender expected %T type, but got: %T", &proto.Command{}, msg.Data())` instead. `Raw()` isn't a function for core.Message.

Verifying logging From test..
`time="2023-05-19T06:45:52-07:00" level=warning msg="metrics sender expected *proto.Command type, but got: *proto.Command_AgentConfig"`